code/game/mecha/mecha.dm

```diff
@@ -1014,9 +1014,9 @@
 			to_chat(occupant, "[icon2html(src, occupant)] [message]")
 	return
 
-/obj/mecha/proc/log_message(message as text,red=null)
+/obj/mecha/log_message(message as text, message_type=LOG_GAME, color=null)

```

Comment:
Won't this conflict with standard logging ? This is for in-game logs isn't it ?

Reply:
Originally both it was, but I tried to make all in-game logs also go to the log files (and vice versa) to keep them consistent with each other. I actually forgot the `..()` in that proc, which was the whole reason I changed the arguments - it needs the `message_type` for `/atom/proc/log_message()`.
Also I changed all of the call sites so that they pass `color="red"` instead of `red=1`, so the original behavior should be preserved.

However the more I look at it the less certain I am about the way BYOND handles arguments, so I may be wrong.

Reply:
Just did some testing with a Gygax, it looks like both the mech's in-game logs and the log files are working fine (after 6be2969).